dll/win32/msgina/gui.c

```diff
@@ -30,14 +31,62 @@ typedef struct _LEGALNOTICEDATA
 typedef struct _DLG_DATA
 {
     PGINA_CONTEXT pgContext;
-    HBITMAP hBitmap;
+    HBITMAP hLogoBitmap;
     HBITMAP hBarBitmap;
     UINT_PTR TimerID;
     DWORD BarCounter;
+    DWORD LogoWidth;
+    DWORD LogoHeight;
     DWORD BarWidth;
     DWORD BarHeight;
 } DLG_DATA, *PDLG_DATA;
 
+static PDLG_DATA
+DlgData_Create(HWND hwndDlg)
+{
+    PDLG_DATA pDlgData;
+    pDlgData = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, sizeof(DLG_DATA));
+    if (pDlgData)
+    {
+        SetWindowLongPtrW(hwndDlg, GWLP_USERDATA, (LONG_PTR)pDlgData);
+    }
+    return pDlgData;
+}
+
+static BOOL
+DlgData_LoadBitmaps(PDLG_DATA pDlgData)
+{
+    BITMAP bm;
+    if (!pDlgData)
+        return FALSE;
+
+    pDlgData->hLogoBitmap = LoadImageW(pDlgData->pgContext->hDllInstance,
+                                       MAKEINTRESOURCEW(IDI_ROSLOGO), IMAGE_BITMAP,
+                                       0, 0, LR_DEFAULTCOLOR);
+    GetObject(pDlgData->hLogoBitmap, sizeof(bm), &bm);
+    pDlgData->LogoWidth = bm.bmWidth;
+    pDlgData->LogoHeight = bm.bmHeight;
+
+    pDlgData->hBarBitmap = LoadImageW(hDllInstance, MAKEINTRESOURCEW(IDI_BAR),
+                                      IMAGE_BITMAP, 0, 0, LR_DEFAULTCOLOR);
+    GetObject(pDlgData->hBarBitmap, sizeof(bm), &bm);
+    pDlgData->BarWidth = bm.bmWidth;
+    pDlgData->BarHeight = bm.bmHeight;
+
+    return pDlgData->hLogoBitmap && pDlgData->hBarBitmap;
```

Comment:
There is some confusion here. Your `DlgData_LoadBitmaps()` is defined as a **BOOL** yet you return two handles of logo and bar image bitmaps. Did I miss something (is it correct in C)?

Reply:
The result will be TRUE if both hLogoBitmap and hBarBitmap are non-NULL. However if any of these is NULL then the returned value will be FALSE.